README.md

```diff
@@ -86,6 +87,73 @@ Arguments are the same as for `CSSselect(query, elems)`. Only returns the first
 - `xmlMode`: When enabled, tag names will be case-sensitive. Default: `false`.
 - `strict`: Limits the module to only use CSS3 selectors. Default: `false`.
 - `rootFunc`: The last function in the stack, will be called with the last element that's looked at. Should return `true`.
+- `adapter`: The adapter to use when interacting with the backing DOM structure. By default it uses [`domutils`](https://github.com/fb55/domutils).
+
+#### Custom Adapters
+
+A custom adapter must implement the following functions:
+
+```
+isTag, existsOne, getAttributeValue, getChildren, getName, getParent,
+getSiblings, getText, hasAttrib, removeSubsets, findAll
+```
+
+The method signature notation used below should be fairly intuitive - if not,
+see the [`rtype`](https://github.com/ericelliott/rtype) or
+[`TypeScript`](https://www.typescriptlang.org/) docs, as it is very similar to
+both of those. You may also want to look at
+[`domutils`](https://github.com/fb55/domutils) to see the default
+implementation.
+
+```ts
```

Comment:
Something like this?

https://gist.github.com/nrkn/b0b096ec4d48b2c590af7ec2301ae5e3